Feat: OkHttp auto instrumentation and Guide (#3298)
getsentry/sentry-docs: src/platforms/android/common/performance/included-instrumentation.mdx

```

When you didn't finish the transaction yet, but you start a new Activity, the SDK always automatically finishes the previous Activity transaction. This is due to the fact that only one transaction at a time can be bound to the Scope. To work around that, you may create transactions manually using the custom instrumentation and its instance to start spans instead of the Static API.

### OkHttp library Instrumentation

> The OkHttp's instrumentation, once added the `SentryOkHttpInterceptor`, starts a Span out of the active Span bound to the Scope for each HTTP Request. The SDK sets the Span `operation` to `http.client` and `description` to Request `$method and $URL`, for example, `GET https://sentry.io`

Reviewer: ```suggestion
The OkHttp's instrumentation, once added the `SentryOkHttpInterceptor`, starts a span out of the active span bound to the scope for each HTTP Request. The SDK sets the span `operation` to `http.client` and `description` to request `$method and $URL`; for example, `GET https://sentry.io`.
```

> The OkHttp's instrumentation, once added the `SentryOkHttpInterceptor`, starts a span out of the active span bound to the scope for each HTTP Request. The SDK sets the span `operation` to `http.client` and `description` to request `$method and $URL`; for example, `GET https://sentry.io`.

Reviewer 2: `m`: If you are new to Sentry and have no clue what a span, an active span, or a scope is, you might have difficulty understanding what this is doing. Can we maybe describe on a higher, more abstract level what this is doing before going into such detail?

Author: to come to this page, people already went thru https://sentry-docs-git-feat-okhttp.sentry.dev/platforms/android/guides/okhttp/performance/ which links to https://sentry-docs-git-feat-okhttp.sentry.dev/product/performance/distributed-tracing/ as well, so I guess we don't need it, it's a subpage of the performance docs

> The OkHttp's instrumentation, once added the `SentryOkHttpInterceptor`, starts a Span out of the running Transaction bound to the Scope for each HTTP Request. The SDK sets the Span `operation` to `http.client` and `description` to Request `$method and $URL`, for example, `GET https://sentry.io`

Author: ```suggestion
The OkHttp's instrumentation, once added the `SentryOkHttpInterceptor`, starts a Span out of the active Span bound to the Scope for each HTTP Request. The SDK sets the Span `operation` to `http.client` and `description` to Request `$method and $URL`, for example, `GET https://sentry.io`
```

> The Span finishes once the Request has been executed, the Span `status` will be depending on the HTTP Response `code` or `SpanStatus.INTERNAL_ERROR` if the `code` does not match any of Sentry's `SpanStatus`.

Reviewer: ```suggestion
The span finishes once the request has been executed. The span `status` depends on either the HTTP Response `code` or `SpanStatus.INTERNAL_ERROR` if the `code` does not match any of Sentry's `SpanStatus`.
```

> When the HTTP Request throws an `IOException`, Sentry's SDK associates this exception to the running Span, in case you don't swallow the exception and capture it, Span and SentryEvent on Sentry dashboard will be linked.

Reviewer: Edited this sentence to clarify:
```suggestion
When the HTTP request throws an `IOException`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and SentryEvent will be linked when viewing it on the *Issue Details** page in sentry.io.
```

> When the HTTP request throws an `IOException`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and SentryEvent will be linked when viewing it on the *Issue Details** page in sentry.io.

Reviewer 2: `l`: I think there is a typo in `*IssueDetails**`

Author: good catch

> When the HTTP request throws an `IOException`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and SentryEvent will be linked when viewing it on the *Issue Details** page in sentry.io.

Author: ```suggestion
When the HTTP request throws an `IOException`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and SentryEvent will be linked when viewing it on the **Issue Details** page in sentry.io.
```

> For more information see our <Link to="/platforms/android/guides/okhttp/">OkHttp</Link> integration.

Reviewer: ```suggestion
For more information see our [OkHttp integration](/platforms/android/guides/okhttp/).
```
